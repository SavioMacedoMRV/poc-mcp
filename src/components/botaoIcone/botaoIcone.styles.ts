import React, {ButtonHTMLAttributes} from 'react'
import styled from 'styled-components'
import {margens} from 'resources/margens'
import {cores} from 'resources/cores'

export interface BotaoProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  icone: React.FC<React.SVGProps<SVGSVGElement>>
  cor?: string
  altura?: number
  largura?: number
  inclinacao?: number
  borda?: boolean
  semMargem?: boolean
  aoClicar: (() => void) | ((e: any) => void)
}

interface StyledProps
  extends Omit<BotaoProps, 'aoClicar' | 'icone' | 'borda' | 'semMargem'> {
  $borda?: boolean
  $semMargem?: boolean
}

export const BotaoStyled = styled.button<StyledProps>`
  display: flex;
  border-radius: ${margens.xsmall}px;
  border: ${({$borda, cor}) =>
    $borda ? `2px solid ${cor ?? cores.neutralDark}` : 'none'};
  align-items: center;
  justify-content: center;
  background: transparent;
  padding: ${({$semMargem}) => ($semMargem ? 0 : margens.xsmall)}px;
  opacity: ${({disabled}) => (disabled ? 0.2 : 1.0)};
  cursor: ${({disabled}) => (disabled ? 'not-allowed' : 'pointer')};
  &:hover {
    opacity: ${({disabled}) => (disabled ? 0.2 : 0.7)};
  }
  &:active {
    opacity: ${({disabled}) => (disabled ? 0.2 : 1)};
  }
  &:focus-visible {
    outline: unset;
    border: 2px solid ${cores.neutralMedium};
  }
`
