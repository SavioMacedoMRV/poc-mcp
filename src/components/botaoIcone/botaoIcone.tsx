import React from 'react'
import {BotaoProps, BotaoStyled} from './botaoIcone.styles'
import {StyleSheetManager} from 'styled-components'
import {cores} from 'resources/cores'

export const BotaoIcone = ({
  aoClicar,
  borda,
  semMargem,
  icone: Icone,
  altura = 32,
  largura = altura,
  inclinacao = 0,
  cor = cores.neutralXDark,
  ...props
}: Readonly<BotaoProps>) => {
  return (
    <StyleSheetManager shouldForwardProp={prop => !!prop}>
      <BotaoStyled
        $borda={borda}
        $semMargem={semMargem}
        cor={cor}
        onClick={aoClicar}
        {...props}
      >
        <Icone
          data-testid={'icone'}
          fill={cor}
          width={largura}
          height={altura}
          style={{transform: `rotate(${inclinacao}deg)`}}
        />
      </BotaoStyled>
    </StyleSheetManager>
  )
}
