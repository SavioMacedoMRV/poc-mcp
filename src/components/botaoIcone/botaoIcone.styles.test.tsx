import {render} from '@testing-library/react'
import * as S from './botaoIcone.styles'

describe('BotaoIcone.styles', () => {
  test('BotaoStyled DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.BotaoStyled />)
    expect(container).toMatchSnapshot()
  })

  test('BotaoStyled DEVE ser igual ao snapshot QUANDO $borda for verdadeiro', () => {
    const {container} = render(<S.BotaoStyled $borda />)
    expect(container).toMatchSnapshot()
  })

  test('BotaoStyled DEVE ser igual ao snapshot QUANDO $borda for verdadeiro e cor', () => {
    const {container} = render(<S.BotaoStyled $borda cor={'red'} />)
    expect(container).toMatchSnapshot()
  })

  test('BotaoStyled DEVE ser igual ao snapshot QUANDO $semMargem for verdadeiro', () => {
    const {container} = render(<S.BotaoStyled $semMargem />)
    expect(container).toMatchSnapshot()
  })
})
