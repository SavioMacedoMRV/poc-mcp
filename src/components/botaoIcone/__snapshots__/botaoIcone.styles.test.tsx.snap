// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BotaoIcone.styles BotaoStyled DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: flex;
  border-radius: 8px;
  border: none;
  align-items: center;
  justify-content: center;
  background: transparent;
  padding: 8px;
  opacity: 1;
  cursor: pointer;
}

.c0:hover {
  opacity: 0.7;
}

.c0:active {
  opacity: 1;
}

.c0:focus-visible {
  outline: unset;
  border: 2px solid #999E9C;
}

<div>
  <button
    class="c0"
  />
</div>
`;

exports[`BotaoIcone.styles BotaoStyled DEVE ser igual ao snapshot QUANDO $borda for verdadeiro 1`] = `
.c0 {
  display: flex;
  border-radius: 8px;
  border: 2px solid #5D615F;
  align-items: center;
  justify-content: center;
  background: transparent;
  padding: 8px;
  opacity: 1;
  cursor: pointer;
}

.c0:hover {
  opacity: 0.7;
}

.c0:active {
  opacity: 1;
}

.c0:focus-visible {
  outline: unset;
  border: 2px solid #999E9C;
}

<div>
  <button
    class="c0"
  />
</div>
`;

exports[`BotaoIcone.styles BotaoStyled DEVE ser igual ao snapshot QUANDO $borda for verdadeiro e cor 1`] = `
.c0 {
  display: flex;
  border-radius: 8px;
  border: 2px solid red;
  align-items: center;
  justify-content: center;
  background: transparent;
  padding: 8px;
  opacity: 1;
  cursor: pointer;
}

.c0:hover {
  opacity: 0.7;
}

.c0:active {
  opacity: 1;
}

.c0:focus-visible {
  outline: unset;
  border: 2px solid #999E9C;
}

<div>
  <button
    class="c0"
    cor="red"
  />
</div>
`;

exports[`BotaoIcone.styles BotaoStyled DEVE ser igual ao snapshot QUANDO $semMargem for verdadeiro 1`] = `
.c0 {
  display: flex;
  border-radius: 8px;
  border: none;
  align-items: center;
  justify-content: center;
  background: transparent;
  padding: 0px;
  opacity: 1;
  cursor: pointer;
}

.c0:hover {
  opacity: 0.7;
}

.c0:active {
  opacity: 1;
}

.c0:focus-visible {
  outline: unset;
  border: 2px solid #999E9C;
}

<div>
  <button
    class="c0"
  />
</div>
`;
